src/circles.rs

```diff
@@ -1,121 +1,131 @@
-            y_center: -a
+            x_center: -2.0 * b,
+            y_center: -a,
         },
         CirclePosition {
             x_center: -b,
-            y_center: -a/2
+            y_center: -a / 2.0,
         },
         CirclePosition {
-            x_center: -2*b,
-            y_center: 0
+            x_center: -2.0 * b,
+            y_center: 0.0,
         },
         CirclePosition {
-            x_center: -2*b,
-            y_center: a
+            x_center: -2.0 * b,
+            y_center: a,
         },
         CirclePosition {
             x_center: -b,
-            y_center: a/2
+            y_center: a / 2.0,
         },
         CirclePosition {
             x_center: -b,
-            y_center: 3*a/2
+            y_center: 3.0 * a / 2.0,
         },
         CirclePosition {
-            x_center: 0,
-            y_center: 2*a
+            x_center: 0.0,
+            y_center: 2.0 * a,
         },
         CirclePosition {
-            x_center: 0,
-            y_center: a
+            x_center: 0.0,
+            y_center: a,
         },
         CirclePosition {
             x_center: b,
-            y_center: 3*a/2
+            y_center: 3.0 * a / 2.0,
         },
         CirclePosition {
-            x_center: 2*b,
-            y_center: a
+            x_center: 2.0 * b,
+            y_center: a,
         },
         CirclePosition {
             x_center: b,
-            y_center: a/2
+            y_center: a / 2.0,
         },
         CirclePosition {
-            x_center: 2*b,
-            y_center: 0
+            x_center: 2.0 * b,
+            y_center: 0.0,
         },
         CirclePosition {
-            x_center: 2*b,
-            y_center: -a
+            x_center: 2.0 * b,
+            y_center: -a,
         },
         CirclePosition {
             x_center: b,
-            y_center: -a/2
+            y_center: -a / 2.0,
         },
         CirclePosition {
             x_center: b,
-            y_center: -3*a/2
+            y_center: -3.0 * a / 2.0,
         },
         CirclePosition {
-            x_center: 0,
-            y_center: 0
-        }
+            x_center: 0.0,
+            y_center: 0.0,
+        },
     ]
 }
 
-/// function to finalize 19 circles with properly corresponding colors and radius
-fn get_colored_circles (center_to_center: i32, small_radius: i32, colors: Vec<[u8; 4]>) -> Vec<Circle> {
+/// Function to finalize 19 circles with properly corresponding colors and radius
+#[cfg(feature = "pix")]
+fn get_colored_circles(
+    center_to_center: f32,
+    small_radius: f32,
+    colors: [[u8; 4]; 19],
+) -> Vec<Circle> {
     let positions = position_circle_set(center_to_center);
     let mut out: Vec<Circle> = Vec::with_capacity(19);
-// no checking is done here for positions.len() == 19 and colors.len() == 19;
-// however, no other length is expected.
+    // no checking is done here for positions.len() == 19 and colors.len() == 19;
+    // however, no other length is expected.
```

Comment:
Adding an `assert` and writing a unit test for any function would be recommended.

Reply:
Changed to arrays with 19 elements instead.